components/script/dom/htmlscriptelement.rs

```diff
@@ -53,6 +63,79 @@ use std::sync::{Arc, Mutex};
 use style::str::{StaticStringVec, HTML_SPACE_CHARACTERS};
 use uuid::Uuid;
 
+pub struct OffThreadCompilationContext {
+    script_element: Trusted<HTMLScriptElement>,
+    script_kind: ExternalScriptKind,
+    final_url: ServoUrl,
+    url: ServoUrl,
+    task_source: NetworkingTaskSource,
+    canceller: TaskCanceller,
+    script_text: String,
+}
+
+/// A wrapper to mark OffThreadToken as Send,
+/// which should be safe according to
+/// mozjs/js/public/OffThreadScriptCompilation.h
+struct OffThreadCompilationToken(Box<OffThreadToken>);
```

Comment:
Why do we need a Box?

Reply:
I got a crash at the point of using the token from inside the task that looked very similar to the crash where you pointed out the need to box the pointer, so I think this is a similar situation?

Reply:
It's not clear to me how a Box actually helps anything here, since the token is just a pointer. I'd like to see what happens if we remove the Box and just have an OffThreadCompilationToken on the stack that gets used in the task in off_thread_compilation_callback.

Actually, it's not clear to me how the current code even works - we're calling Box::from_raw on the token argument, but we never created it from Box::into_raw so that's invalid.

Reply:
Ok, it's worth a try to see if the boxing is unnecessary @AbhishekSharma102. By the way I think it is valid to use `from_raw` on a pointer that wasn't created via `into_raw`, see the second example at https://doc.rust-lang.org/std/boxed/struct.Box.html#method.from_raw

Reply:
You are correct that it can be valid if it was created by a Rust allocator. The OffThreadToken is created by the spidermonkey allocator, though.

Reply:
Ok, thanks for the info.

Reply:
The docs do explain this in more detail over at https://doc.rust-lang.org/std/boxed/index.html#memory-layout (which is linked to from the `from_raw` docs)

Reply:
I am having trouble running wpt locally to test this out. Unit tests are passing without Box as well (i guess that happened last time as well, wpts failed while ut passed)

Reply:
> I am having trouble running wpt locally to test this out.

What problem are you encountering? 

If you don't know exactly which test to run, I think any test will do, since all will load scripts I believe. You could try for example `./mach test-wpt tests/wpt/web-platform-tests/html/`(maybe with the `--release` flag depending on what you have build locally). 

If you don't get a crash, I think it's worth pushing your changes and we can try here as well.

Reply:
I keep getting timeouts, probably due to low sku.

Reply:
yes I think that's normal, but most are ok right? Or does everything time-out? 

Sounds like it's working, and maybe you can address the other comments if not already, and then push the changes and we can take it from there?

Reply:
Apart from CompileOffThread1, I've addressed the rest.

 Will push after rebasing.

Reply:
when i do ./mach build --dev, it tries to update a few things and demands for almost 52GB of memory, and fails.
memory allocation of 52834435104 bytes failed.

Isnt that a bit too much?